### PR TITLE
CSHARP-5237: Remove optional MD5 support from GridFS

### PR DESCRIPTION
**Pull request by papafe**

### CODE REVIEW

File: src/MongoDB.Driver/GridFS/GridFSForwardOnlyDownloadStream.cs [hunk 1]
```diff
@@ -50,16 +48,9 @@ internal class GridFSForwardOnlyDownloadStream<TFileId> : GridFSDownloadStreamBa
         public GridFSForwardOnlyDownloadStream(
             GridFSBucket<TFileId> bucket,
             IReadBinding binding,
-            GridFSFileInfo<TFileId> fileInfo,
-            bool checkMD5)
+            GridFSFileInfo<TFileId> fileInfo)
```
Comment (BorisDog) [inline src/MongoDB.Driver/GridFS/GridFSForwardOnlyDownloadStream.cs:51]: Do we also need remove MD5 option in `GridFSForwardOnlyUploadStream`,`GridFSBucketOptions`, `GridFSFileInfoCompat` and `GridFSUploadOptions`?

**Review by BorisDog — CHANGES_REQUESTED** [review]
Could you also create a ticket to mark this API as obsolete in 2.x please?

File: src/MongoDB.Driver/GridFS/GridFSForwardOnlyDownloadStream.cs (hunk 1 above)
Comment (papafe) [reply src/MongoDB.Driver/GridFS/GridFSForwardOnlyDownloadStream.cs:51]: Yes, sorry, I should have opened this as a draft

**Review by BorisDog — APPROVED** [review]
LGTM!

**Pull request merged**